### DISCUSSION TITLE
Micrometer support for fully qualified metrics names

### DISCUSSION THREAD

**Discussion by arpit-gupta213** [quarkusio/quarkus#27983] [Q&A]
I am trying to migrate from microprofile to micrometer metrics in my project. The current implementation uses the fully qualified metrics names. For example, if I have a metrics with name "totalCount" in the class com.org.github.MicrometerMetrics the way I am accessing the metrics is by using the key: "**com.org.github.MicrometerMetrics.totalCount**"
When I switched to microprofile the default names of metrics changes from the fully qualified names to just the metrics name. Meaning now I have to access the metrics using the key: "**totalCount**"
This makes the migration complex as I would have to change the name of the metrics to make them meaningful without the fully qualified name and also I have to change the key to access the metrics

Is there a way in micrometer to get the metrics by fully qualified names which includes package name and class name along with the metrics name?

[a comment by an automated account is not shown]

**Comment by ebullient** [comment]
The short answer is no. 

Are you using MP annotations, or looking for built in metrics?  While MP Metrics does use fully qualified classnames + method names, it is true that Micrometer does not (unless you configure it that way). 

General conventions for metrics are not class/package name bound. Consider the case that you made another class that serves a similar purpose but uses slightly different method names and wanted to aggregate across those... Micrometer makes a best effort at setting you down that path. I can look at adding the full class name as a tag..

**Reply by arpit-gupta213** [reply]
Thanks for clarification. I am using MP annotations for the metrics. My aim is to have minimum changes while migrating from Microprofile to Micrometer. Changing each metrics names to their fully qualified names (classnames + method names) is something I wanted to avoid. I do feel that fully qualified names are not desirable for metrics but since I was using Microprofile and all the places where I am reading metrics values are configured with fully qualified metrics name, it is time consuming to make changes everywhere to now use just the metrics names.

Just wanted to check if we have any out of the box configuration in micrometer which enables the fully qualified metrics names.

**Reply by ebullient** [reply]
If the short method name is "unique".. you could use a MeterFilter to put the class name back in..(renaming with MeterFilter is usually the way to go, but if you have identically named methods, that doesn't work as well). If I add an option to add the classname as a tag.. a meter filter could be used to smoosh the label value back into the metric name..
